public/docs/_examples/server-communication/dart/lib/wiki/wiki_smart_component.dart

```diff
@@ -0,0 +1,36 @@
+// #docregion
+import 'package:angular2/angular2.dart';
+import 'wikipedia_service.dart';
+import 'package:stream_transformers/stream_transformers.dart';
+
+@Component(
+    selector: 'my-wiki-smart',
+    template: '''
+      <h1>Smarter Wikipedia Demo</h1>
+      <p><i>Fetches when typing stops</i></p>
+
+      <input #term (keyup)="search(term.value)"/>
+      <ul>
+        <li *ngFor="#item of items">{{item}}</li>
+      </ul>
+    ''',
+    providers: const [WikipediaService])
+class WikiSmartComponent {
+  final WikipediaService _wikipediaService;
+  List items;
```

Comment:
Prefer initializing items as in general you might get unexpected behaviour as things will be null for a brief moment.